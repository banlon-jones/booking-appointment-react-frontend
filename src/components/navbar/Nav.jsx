import React, { useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { logOutUser } from '../../store/user/user';
import './Nav.css';

const Nav = () => {
  const dispatch = useDispatch();
  const [isMobile, setIsMobile] = useState(false);
  const { loggedIn } = useSelector((state) => state.user);
  const { role } = useSelector((state) => state.user);

  return (
    <div className="navigation-panel">
      <header>
        <div
          className={
            isMobile
              ? 'nav-icons-holder '
              : 'nav-icons-holder nav-mobile-border'
          }
        >
          <h4>Resort Booking</h4>
          <div className="nav-btn-holder">
            {!loggedIn && (
              <>
                <Link to="/login">
                  <button className="button-sign-log" type="button">
                    Log in
                  </button>
                </Link>
                <Link to="/sign-up">
                  <button className="button-sign-log" type="button">
                    Sign up
                  </button>
                </Link>
              </>
            )}
            {loggedIn && (
              <button
                className="button-icon"
                type="submit"
                onClick={() => setIsMobile(!isMobile)}
              >
                {isMobile ? (
                  <i className="fas fa-times fa-2x" />
                ) : (
                  <i className="fas fa-bars fa-2x" />
                )}
              </button>
            )}
          </div>
        </div>
      </header>
      {loggedIn && (
        <nav>
          <ul className={isMobile ? 'show-mobile-menu' : 'hide-mobile-menu'}>
            <li className="nav-list">
              <Link
                className="nav-link link-style"
                to="/resorts"
                onClick={() => setIsMobile(!isMobile)}
              >
                Resorts
              </Link>
            </li>
            <li className="nav-list">
              <Link
                className="nav-link link-style"
                to="/reserve"
                onClick={() => setIsMobile(!isMobile)}
              >
                Reserve
              </Link>
            </li>
            <li className="nav-list">
              <Link
                className="nav-link link-style"
                to="/reservations"
                onClick={() => setIsMobile(!isMobile)}
              >
                My reservations
              </Link>
            </li>
            {role === 'admin' && (
              <>
                <li className="nav-list">
                  <Link
                    className="nav-link link-style"
                    to="/addItem"
                    onClick={() => setIsMobile(!isMobile)}
                  >
                    Add resort
                  </Link>
                </li>
                <li className="nav-list">
                  <Link
                    className="nav-link link-style"
                    to="/deleteResorts"
                    onClick={() => setIsMobile(!isMobile)}
                  >
                    Delete resort
                  </Link>
                </li>
              </>
            )}
            <li>
              <button
                className="button-sign-logout"
                type="button"
                onClick={() => dispatch(logOutUser())}
              >
                Log out
              </button>
            </li>
          </ul>
        </nav>
      )}
      <footer className={isMobile ? 'show-mobile-menu' : 'hide-mobile-menu'}>
        <ul className="footer-ul ul-style">
          <li className="social-media-list">
            <Link
              className="social-media-link link-style"
              to="https://twitter.com/"
            >
              <i className="fab fa-twitter" />
            </Link>
          </li>
          <li className="social-media-list">
            <Link
              className="social-media-link link-style"
              to="https://www.facebook.com"
            >
              <i className="fab fa-facebook-f" />
            </Link>
          </li>
          <li className="social-media-list">
            <Link
              className="social-media-link link-style"
              to="https://www.google.com/"
            >
              <i className="fab fa-google-plus-g" />
            </Link>
          </li>
          <li className="social-media-list">
            <Link
              className="social-media-link link-style"
              to="https://vine.co/"
            >
              <i className="fab fa-vine" />
            </Link>
          </li>
          <li className="social-media-list">
            <Link
              className="social-media-link link-style"
              to="https://www.pinterest.com/"
            >
              <i className="fab fa-pinterest-p" />
            </Link>
          </li>
        </ul>
      </footer>
    </div>
  );
};

export default Nav;
