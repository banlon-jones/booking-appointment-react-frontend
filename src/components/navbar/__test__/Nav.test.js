import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import Nav from '../Nav';

describe('snapshots', () => {
  const initialState = { user: { loggedIn: false, role: '' } };
  const mockStore = configureStore();
  const store = mockStore(initialState);

  it('matches LogIn snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Nav />
          </Router>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
