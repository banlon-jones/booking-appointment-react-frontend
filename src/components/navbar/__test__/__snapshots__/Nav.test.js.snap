// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots matches LogIn snapshot 1`] = `
<div
  className="navigation-panel"
>
  <header>
    <div
      className="nav-icons-holder nav-mobile-border"
    >
      <img
        alt="Logo"
        className="logo"
        src="logo.jpg"
      />
      <div
        className="nav-btn-holder"
      >
        <a
          href="/login"
          onClick={[Function]}
        >
          <button
            className="button-sign-log"
            type="button"
          >
            Log in
          </button>
        </a>
        <a
          href="/sign-up"
          onClick={[Function]}
        >
          <button
            className="button-sign-log"
            type="button"
          >
            Sign up
          </button>
        </a>
      </div>
    </div>
  </header>
  <footer
    className="hide-mobile-menu"
  >
    <ul
      className="footer-ul ul-style"
    >
      <li
        className="social-media-list"
      >
        <a
          className="social-media-link link-style"
          href="/https://twitter.com/"
          onClick={[Function]}
        >
          <i
            className="fab fa-twitter"
          />
        </a>
      </li>
      <li
        className="social-media-list"
      >
        <a
          className="social-media-link link-style"
          href="/https://www.facebook.com"
          onClick={[Function]}
        >
          <i
            className="fab fa-facebook-f"
          />
        </a>
      </li>
      <li
        className="social-media-list"
      >
        <a
          className="social-media-link link-style"
          href="/https://www.google.com/"
          onClick={[Function]}
        >
          <i
            className="fab fa-google-plus-g"
          />
        </a>
      </li>
      <li
        className="social-media-list"
      >
        <a
          className="social-media-link link-style"
          href="/https://vine.co/"
          onClick={[Function]}
        >
          <i
            className="fab fa-vine"
          />
        </a>
      </li>
      <li
        className="social-media-list"
      >
        <a
          className="social-media-link link-style"
          href="/https://www.pinterest.com/"
          onClick={[Function]}
        >
          <i
            className="fab fa-pinterest-p"
          />
        </a>
      </li>
    </ul>
  </footer>
</div>
`;
